subsys/bluetooth/host/hci_core.c

```diff
@@ -228,18 +228,59 @@ static void handle_vs_event(uint8_t event, struct net_buf *buf,
 	/* Other possible errors are handled by handle_event_common function */
 }
 
-#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
-void bt_hci_host_num_completed_packets(struct net_buf *buf)
+enum acl_data_flags *bt_get_acl_buf_flags(struct net_buf *acl_packet)
 {
+	return &(acl(acl_packet)->flags);
+}
 
+void bt_send_one_host_num_completed_packets(uint16_t handle)
+{
+#if !defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ARG_UNUSED(handle);
+	return;
+#else
 	struct bt_hci_cp_host_num_completed_packets *cp;
-	uint16_t handle = acl(buf)->handle;
 	struct bt_hci_handle_count *hc;
+	struct net_buf *buf;
+	int err;
+
+	LOG_DBG("Reporting completed packet for handle %u", handle);
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_HOST_NUM_COMPLETED_PACKETS,
+				sizeof(*cp) + sizeof(*hc));
+	if (!buf) {
+		LOG_ERR("Unable to allocate new HCI command");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->num_handles = sys_cpu_to_le16(1);
+
+	hc = net_buf_add(buf, sizeof(*hc));
+	hc->handle = sys_cpu_to_le16(handle);
+	hc->count  = sys_cpu_to_le16(1);
+
+	err = bt_hci_cmd_send(BT_HCI_OP_HOST_NUM_COMPLETED_PACKETS, buf);
+	if (err) {
+		LOG_ERR("Unable to command 😭");
+	}
```

Comment:
```suggestion
	if (err) {
		LOG_ERR("Unable to send host num completed packes");
	}
```

Is this something we can recover from in any way?

Reply:
I don't think so. If the HCI driver returned an error sending a command, that probably means the transport is messed up. Apart from disabling bluetooth, I can't think of anything.